### PR TITLE
chore: Merge master to develop and manage conflict

### PR DESCRIPTION
**Pull request by adrien2p**
FIXES CORE-1278

### CODE REVIEW

**Review by shahednasser — APPROVED** [review]
Docs LGTM

**Comment by olivermrbl** [comment]
@adrien2p I would like us to handle the documentation merge in a different PR, as it clutters this one.

Also, I'd like @shahednasser to be responsible for that :)

**Comment by adrien2p** [comment]
No problem @olivermrbl i can revert all documentation related changes 👌

**Comment by adrien2p** [comment]
@fPolic could I get you to look at the new hook usage to be sure I did not insert an issue :) it should be fine but just a second pair of eyes just in case 👍

File: packages/medusa/src/services/product-variant.ts [hunk 1]
```diff
@@ -343,34 +450,256 @@ class ProductVariantService extends TransactionBaseService {
...
+
+      if (dataCurrencyPrices.length) {
+        promises.push(this.addOrUpdateCurrencyPrices(dataCurrencyPrices))
+      }
+
+      await Promise.all(promises)
+    })
+  }
+
+  async addOrUpdateRegionPrices(
```
Comment (fPolic) [inline packages/medusa/src/services/product-variant.ts:543]:
```suggestion
  async upsertRegionPrices(
```

**Review by fPolic — APPROVED** [review]
LGTM @adrien2p 💪

**Pull request merged**